### PR TITLE
Feature/update

### PR DESCRIPTION
**Pull request by kewlfft**
pull request after correction have been made

### CODE REVIEW

File: aur.py [hunk 1]
```diff
@@ -15,6 +15,18 @@ def package_installed(module, package_name):
...
 
 
+def update_packages(module, tool):
+    assert tool in TOOL_TO_INSTALL_CMD_MAP
+
+    cmd = TOOL_TO_INSTALL_CMD_MAP[tool] + ['--update']
+    rc, stdout, stderr = module.run_command(cmd, check_rc=True)
+
+    module.exit_json(
+        changed=stdout.count('\n') > 2,
```
Comment (cdown) [inline aur.py]: This is really quite fragile, just set the locale to C

**Review by cdown — CHANGES_REQUESTED** [review]
Looks ok pending comment

File: aur.py (hunk 1 above)
Comment (kewlfft) [reply aur.py]: ok will do, thanks for the feedback

**Comment by kewlfft** [comment]
locale set to C as suggested, please let me know if acceptable now

File: aur.py [hunk 2]
```diff
@@ -15,6 +15,18 @@ def package_installed(module, package_name):
     return exit_code == 0
 
 
+def update_packages(module, tool):
+    assert tool in TOOL_TO_INSTALL_CMD_MAP
+
+    cmd = ['env', 'LC_ALL=C'] + TOOL_TO_INSTALL_CMD_MAP[tool] + ['--update']
```
Comment (cdown) [inline aur.py]:
```
% yaourt --update
pacman: unrecognized option '--update'
```

**Review by cdown — CHANGES_REQUESTED** [review]

File: aur.py (hunk 2 above)
Comment (kewlfft) [reply aur.py]: yes indeed tested with pacaur but for yaourt i need to change the syntax

**Comment by kewlfft** [comment]
used a different syntax to be compatible with both yaourt and pacaur

**Comment by kewlfft** [comment]
however I limit the update to AUR because pacman is expected to be used instead for a full system upgrade, do you think we should leave it open and not limit the update to AUR?
In this case I would remove the --aur option

**Comment by kewlfft** [comment]
or better we can add an AUR argument to limit both searches and updates to AUR or not, I will propose this

**Comment by kewlfft** [comment]
I am leaving the parameter -S / -Su within the functions, this will allow more flexibility for future integration of other helpers

**Comment by kewlfft** [comment]
yay added and tested, yay does not support --aur for the moment but it does not trigger any error in the command line (we could just mention noaur is not supported by yay somewhere), the rest should be fine.

**Comment by cdown** [comment]
Please keep one type of change to one pull request :-) If you want to add update support, do it here. Don't overcomplicate it by adding `yay`/`tizen`/etc at the same time.

File: aur.py [hunk 3]
```diff
@@ -15,16 +16,32 @@ def package_installed(module, package_name):
...
             changed=False,
             msg='package already installed',
         )
 
-    assert tool in TOOL_TO_INSTALL_CMD_MAP
+    assert tool in TOOL_CMD_MAP
 
-    cmd = TOOL_TO_INSTALL_CMD_MAP[tool] + [package_name]
+    cmd = TOOL_CMD_MAP[tool] + ['-S', package_name]
+    if auronly:
```
Comment (cdown) [inline aur.py]: This is implied by calling the `aur` handler.

File: aur.py [hunk 4]
```diff
@@ -61,15 +78,15 @@ def main():
...
+                'required': False,
             },
             'state': {
                 'default': 'present',
                 'choices': ['present', 'absent'],
             },
             'tool': {
                 'default': 'pacaur',
-                'choices': ['pacaur', 'yaourt'],
+                'choices': ['pacaur', 'yaourt', 'yay'],
```
Comment (cdown) [inline aur.py]: This should go in a separate PR.

**Review by cdown — CHANGES_REQUESTED** [review]

File: aur.py (hunk 4 above)
Comment (kewlfft) [reply aur.py]: You are right I will revert these changes

File: aur.py (hunk 3 above)
Comment (kewlfft) [reply aur.py]: What do you recommend ? That the auronly option is set to true by default ?

Comment (cdown) [reply aur.py]: Yeah, it should be the default.

**Comment by kewlfft** [comment]
I removed yay
Regarding the 'auronly' option: -S without --aur looks in repo before trying AUR, adding 'auronly' as an option only installs from the AUR. Therefore I think this is a valid option to be added.

File: aur.py (hunk 3 above)
Comment (kewlfft) [reply aur.py]: ok if I change auronly to default True do you think my proposal is acceptable?

File: aur.py [hunk 5]
```diff
@@ -79,16 +95,27 @@ def main():
...
+            'update': {
+                'default': False,
+                'type': 'bool',
+            },
+            'auronly': {
+                'default': True,
+                'type': 'bool',
+            },
         },
+        required_one_of=[['name', 'update']],
```
Comment (cdown) [inline aur.py:112]: As far as I know this means that if you were to pass both, `name` will simply be ignored, but it should be more obvious that this is invalid

File: README.md [hunk 6]
```diff
@@ -33,6 +33,11 @@ Currently, we support the following AUR helpers:
    - aur: name=yturl tool=yaourt
      [...]
 
+   # Update (using pacaur)
+   - aur: update=yes auronly=yes
+     become:yes
```
Comment (cdown) [inline README.md]: nit: space here

**Review by cdown — CHANGES_REQUESTED** [review]
Looks good to me pending final comments

File: aur.py (hunk 5 above)
Comment (kewlfft) [reply aur.py:112]: I can implement update for individual name it would be cleaner, personally I dont use it but this should be allowed

**Comment by kewlfft** [comment]
I have hopefully addressed your various points, thanks

**Comment by kewlfft** [comment]
Let me know if you think something needs to be changed

**Comment by cdown** [comment]
Squashed/merged/linted as 2cab329ab45b26339e8ddc66046550509cdd40f8. Thanks for the PR!

**Comment by kewlfft** [comment]
Thanks for accepting the pr
